src/main/java/requestMgmt/RequestManager.java

```diff
@@ -0,0 +1,323 @@
+            } else {
+                this.invalidResponseHandler.handleInvalidResponse(status, "Error subscribing to project: " + id);
+            }
+        }, this.requestSendErrorHandler);
+        wsManager.sendAuthenticatedRequest(subscribeRequest);
+    }
+
+    /**
+     * Finish the subscription action by using the project's files from the server and creating metadata.
+     * @param id id of the project subscribed to
+     * @param files files from the server
+     */
+    public abstract void finishSubscribeToProject(long id, File[] files);
+
+    /**
+     * Unsubscribe from the project with the given id
+     * @param id id of the project to unsubscribe from
+     */
+    public void unsubscribeFromProject(long id) {
+        Request request = (new ProjectUnsubscribeRequest(id)).getRequest(response -> {
+            int status = response.getStatus();
+            if (status == 200) {
+                dataManager.getSessionStorage().setUnsubscribed(id);
+            } else {
+                this.invalidResponseHandler.handleInvalidResponse(status, "Failed to unsubscribe from project: " + id);
+            }
+        }, this.requestSendErrorHandler);
+        wsManager.sendAuthenticatedRequest(request);
+    }
+
+    /**
+     * Fetch all projects and subscribe to the set of projects with the given ids.
+     * @param ids the ids of the projects to subscribe to
+     */
+    public void fetchAndSubscribeAll(List<Long> ids) {
+        Request getProjectsRequest = new UserProjectsRequest().getRequest(response2 -> {
+            int status2 = response2.getStatus();
+            if (status2 == 200) {
+                List<Project> projects =
+                        Arrays.asList(((UserProjectsResponse) response2.getData()).getProjects());
+                sendProjectsLookupRequest(projects);
+                for (long id : ids) {
+                    subscribeToProject(id);
+                }
+            } else {
+                this.invalidResponseHandler.handleInvalidResponse(status2, "Error fetching projects");
+            }
+        }, requestSendErrorHandler);
+        this.wsManager.sendAuthenticatedRequest(getProjectsRequest);
+    }
+
+    /**
+     * Create a project with the given name
+     * @param projectName name of the project to create
+     */
+    public void createProject(String projectName) {
+        Request request = (new ProjectCreateRequest(projectName)).getRequest(response -> {
+            int status = response.getStatus();
+            if (status == 200) {
+                long pid = ((ProjectCreateResponse) response.getData()).getProjectID();
+                Request request2 = (new ProjectSubscribeRequest(pid)).getRequest(response2 -> {
+                    int status2 = response2.getStatus();
+                    if (status2 == 200) {
+                        Long[] ids = {pid};
+                        Request request3 = (new ProjectLookupRequest(ids)).getRequest(response3 -> {
```

Comment:
why do you need to lookup a project that you just made? can't you infer everything it would have already? won't it just have your own permission set to 10 and that's it?